### DISCUSSION TITLE
[@types/node] Missing all Web Crypto API

### DISCUSSION THREAD

**Discussion by loky-lp** [DefinitelyTyped/DefinitelyTyped#61211] [Issues with a @types package] [answered]
I'm using the SubtleCrypto API on a project to generate custom JWT signed with RS256 algorithm and by doing so i found out that the whole webcrypto class is missing from the type definitions.

Since this feature was added in both v15 and v16 (now on LTS) not having those types breaks the project building building process of the tools that i'm working now.
I could use `// @ts-ignore` to get it working (because the underlying API code exist) but thats feels like sealing a wound with duct tape.

Example of the error
```typescript
// TS2339: Property 'subtle' does not exist on type 'typeof webcrypto'.
const key = await webcrypto.subtle.importKey(
	"jwk", jwk,
	{
		name: 'RSASSA-PKCS1-v1_5',
		hash: 'SHA-256',
	},
	false, [ "sign" ]
)
```

Offical Node.js [doc](https://nodejs.org/docs/latest-v16.x/api/crypto.html#cryptowebcrypto), Web Crypto API [doc](https://nodejs.org/docs/latest-v16.x/api/webcrypto.html)

[a comment by an automated account is not shown]

**Comment by yoursunny** [comment]
In my package, I have a file that re-exports `crypto` object.

When [targeting Node.js](https://github.com/yoursunny/NDNts/blob/4447a0a31fa9d41ffbd77542cb17e45f15aea0c9/packages/util/src/platform_node.ts#L10-L11), it's:

```js
import { webcrypto } from "node:crypto";

/** Web Crypto API. */
export const crypto: Crypto = webcrypto as any;
```

When [targeting browsers](https://github.com/yoursunny/NDNts/blob/4447a0a31fa9d41ffbd77542cb17e45f15aea0c9/packages/util/src/platform_browser.ts#L2), it's:

```js
export const crypto = globalThis.crypto;
```

Every other file that needs to use WebCrypto would write:

```js
import { crypto } from "./platform";
```

This restricts `as any` to only one place, while supports both Node.js and browser environments.

**Comment by loky-lp** [comment, marked as answer]
PR https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60778 just merged few hours ago, this solves all problems
Also closing issue https://github.com/DefinitelyTyped/DefinitelyTyped/issues/61212
